pkg/provisioner/cloudprovisioner.go

```diff
@@ -585,7 +585,8 @@ func (c *CloudProvisioner) ListSnapshots(
 	// 4. StartingToken is not null, and MaxEntries is not null. Return `MaxEntries` snapshots from `StartingToken`.
 	start := 0
 	if startingToken != "" {
-		start, err := strconv.Atoi(startingToken)
+		var err error
```

Comment:
Any particular reason for changing the declaration?

Reply:
Yes. Using `:=` redeclared another instance of the `start` variable within the `if` statement scope. Any updates within this scope would be lost, and so `ListSnapshots` would always return results from the beginning.